src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -0,0 +1,378 @@
+		value.putAll(speproperties.getEsb());
+		// if no user that use default values (for testing).
+		if (user == null || user.isEmpty()) {
+			value.putAll(speproperties.getPutgrades());
+		}
+		else {
+			value.put("SCORE",(String) user.get("Score"));
+			value.put("UNIQNAME",(String) user.get("Unique_Name"));
+		}
+		return value;
+	}
+
+	//{"Meta":{"Message":"COMPLETED","httpStatus":200},"Result":{"putPlcExamScoreResponse":{"putPlcExamScoreResponse":{"Status":"SUCCESS","Form":7,"ID":"abc"},"@schemaLocation":"http://mais.he.umich.edu/schemas/putPlcExamScoreResponse.v1 http://csqa9ib.dsc.umich.edu/PSIGW/PeopleSoftServiceListeningConnector/putPlcExamScoreResponse.v1.xsd"}}}
+	
+	//{"putPlcExamScoreResponse":{"putPlcExamScoreResponse":{"Status":"SUCCESS","Form":7,"ID":"abc"
+	
+	// Print out user for updated grade.
+	static protected void logPutGrade(WAPIResultWrapper putGrade)  {
+		try {
+			JSONObject jo = new JSONObject(putGrade);
+			JSONObject jAI = jo.getJSONObject("result");
+			JSONObject pPESR = jAI.getJSONObject("putPlcExamScoreResponse").getJSONObject("putPlcExamScoreResponse");
+			String user = pPESR.getString("ID");
+			M_log.info("putGrade: updated MPathways for user: "+user);
+		} catch (JSONException e) {
+			M_log.error("error extracting grade update user for: "+putGrade.toJson());
+		}
+	}
+
+	/********* put in the grades **********/
+	
+	// Send a list of grades to MPathways.
+	public void putSPEGrades(ArrayList<HashMap<String, String>> SPEgradeMaps) {
+		int gradesAdded = 0;
+		int gradesAttempted = 0;
+		Boolean success;
+		for(HashMap<String, String> singleUser : SPEgradeMaps) {
+			gradesAttempted ++;
+			success = putSPEGrade(singleUser);
+			if (success) {
+				gradesAdded++;
+			}
+		}
+		M_log.info("adding grades: attempted: {} successful: {}",gradesAttempted,gradesAdded);
+	}
+	
+	// Send a single grade to MPathways.
+	public boolean putSPEGrade(HashMap<?, ?> user) {
+		HashMap<String,String> value = setupPutGradeCall(user);
+
+		WAPIResultWrapper wrappedResult = speesb.putGradeViaESB(value);
+
+		M_log.debug("update: {}",wrappedResult.toJson());
+
+		if (wrappedResult.getStatus() == 200) {
+			logPutGrade(wrappedResult);
+			return true;
+		}
+
+		M_log.error("error updating grade: "+wrappedResult.toJson());
+		return false;
+	}
+
+
+
+	/***************** manage the timestamp data with PersistString ***********/
+	/*
+	 * Use PersistString class to get a string with a timestamp representing the 
+	 * last time the script updated the grades.
+	 * 
+	 * If the value isn't available or is corrupt use a default value.  We'll be careful
+	 * but won't panic about only submitting grades once.
+	 */
+	
+	// Format an internal timestamp into the expected string.
+	protected static String formatPersistTimestamp(LocalDateTime ts) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+		String s = ts.format(formatter);
+		return s;
+	}
```

Comment:
May be this method should go in a Utility class, that way Each class retains identity of what is is doing.